pkg/tsdb/influxdb/response_parser_test.go

```diff
@@ -1,201 +1,337 @@
 
 		query = &Query{Alias: "alias [[tag_cluster-name]]"}
 		result = parser.Parse(response, query)
 
-		require.Equal(t, result.Series[0].Name, "alias Cluster")
+		require.Equal(t, decodedFrames(t, &result)[0].Name, "alias Cluster")
 
 		query = &Query{Alias: "alias [[tag_/cluster/name/]]"}
 		result = parser.Parse(response, query)
 
-		require.Equal(t, result.Series[0].Name, "alias Cluster/")
+		require.Equal(t, decodedFrames(t, &result)[0].Name, "alias Cluster/")
 
 		query = &Query{Alias: "alias [[tag_@cluster@name@]]"}
 		result = parser.Parse(response, query)
 
-		require.Equal(t, result.Series[0].Name, "alias Cluster@")
+		require.Equal(t, decodedFrames(t, &result)[0].Name, "alias Cluster@")
 	})
 
 	t.Run("Influxdb response parser with errors", func(t *testing.T) {
 		parser := &ResponseParser{}
 
-		cfg := setting.NewCfg()
-		err := cfg.Load(&setting.CommandLineArgs{
-			HomePath: "../../../",
-		})
-		require.Nil(t, err)
-
-		response := &Response{
-			Results: []Result{
+		response := responseFromJSON(t, `
+		{
+			"results": [
 				{
-					Series: []Row{
+					"series": [
 						{
-							Name:    "cpu",
-							Columns: []string{"time", "mean", "sum"},
-							Tags:    map[string]string{"datacenter": "America"},
-							Values: [][]interface{}{
-								{json.Number("111"), json.Number("222"), json.Number("333")},
-								{json.Number("111"), json.Number("222"), json.Number("333")},
-								{json.Number("111"), json.Number("null"), json.Number("333")},
-							},
-						},
-					},
+							"name": "cpu",
+							"columns": ["time","mean","sum"],
+							"tags": {"datacenter": "America"},
+							"values": [
+								[111,222,333],
+								[111,222,333],
+								[111,null,333]
+							]
+						}
+					]
 				},
 				{
-					Err: fmt.Errorf("query-timeout limit exceeded"),
-				},
-			},
+					"error": "query-timeout limit exceeded"
+				}
+			]
 		}
+		`)
 
 		query := &Query{}
 
 		result := parser.Parse(response, query)
 
-		require.Len(t, result.Series, 2)
+		decoded := decodedFrames(t, &result)
+
+		require.Len(t, decoded, 2)
 
-		require.Len(t, result.Series[0].Points, 3)
-		require.Len(t, result.Series[1].Points, 3)
+		require.Equal(t, decoded[0].Fields[0].Len(), 3)
+		require.Equal(t, decoded[0].Fields[1].Len(), 3)
+		require.Equal(t, decoded[1].Fields[0].Len(), 3)
+		require.Equal(t, decoded[1].Fields[1].Len(), 3)
 
 		require.Error(t, result.Error)
 		require.Equal(t, result.Error.Error(), "query-timeout limit exceeded")
 	})
+
+	t.Run("Influxdb response parser with top-level error", func(t *testing.T) {
+		parser := &ResponseParser{}
+
+		response := responseFromJSON(t, `
+		{
+			"error": "error parsing query: found THING"
+		}
+		`)
+
+		query := &Query{}
+
+		result := parser.Parse(response, query)
+
+		require.Nil(t, result.Dataframes)
+
+		require.Error(t, result.Error)
+		require.Equal(t, result.Error.Error(), "error parsing query: found THING")
```

Comment:
Could merge these two into:

```suggestion
	require.EqualError(t, result.Error, "error parsing query: found THING")
```

not your fault, but please if you feel like it, fix the same issue on lines 315-316! Thanks!